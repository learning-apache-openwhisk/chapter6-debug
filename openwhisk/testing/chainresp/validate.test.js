const ow = require('openwhisk')

test('validate', () =>
  ow().actions
    .invoke({
      name: 'testing/chainresp-validate',
      params: {
        name: 'Michele',
        email: 'michele.sciabarra.com',
        phone: '1234567890'
      }
    })
    .then(res => expect(res).toMatchSnapshot()))


test('validate0', () =>
  ow().actions
    .invoke({
      name: 'testing/chainresp-validate',
      params: {}
    })
    .then(res => expect(res).toMatchSnapshot()))

test('validate1', () =>
  ow().actions
    .invoke({
      name: 'testing/chainresp-validate',
      params: { name: 'Michele' }
    })
    .then(res => expect(res).toMatchSnapshot()))

test('validate2', () =>
  ow().actions
    .invoke({
      name: 'testing/chainresp-validate',
      params: {
        name: 'Michele',
        email: 'michele@sciabarra.com'
      }
    })
    .then(res => expect(res).toMatchSnapshot()))

test('validate3', () =>
  ow().actions
    .invoke({
      name: 'testing/chainresp-validate',
      params: {
        name: 'Michele',
        email: 'michele@sciabarra.com',
        phone: '1234567890'
      }
    })
    .then(res => expect(res).toMatchSnapshot()))

