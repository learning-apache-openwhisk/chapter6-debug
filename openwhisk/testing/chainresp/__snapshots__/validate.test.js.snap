// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate 1`] = `
Object {
  "email": "michele.sciabarra.com",
  "errors": Array [
    "michele.sciabarra.com is not an email address",
  ],
  "message": Array [
    "name: Michele",
    "phone: 1234567890",
  ],
  "name": "Michele",
  "phone": "1234567890",
}
`;

exports[`validate0 1`] = `
Object {
  "errors": Array [
    "missing name",
    "missing email",
    "missing phone",
  ],
  "message": Array [],
}
`;

exports[`validate1 1`] = `
Object {
  "errors": Array [
    "missing email",
    "missing phone",
  ],
  "message": Array [
    "name: Michele",
  ],
  "name": "Michele",
}
`;

exports[`validate2 1`] = `
Object {
  "email": "michele@sciabarra.com",
  "errors": Array [
    "michele@sciabarra.com is not an email address",
    "missing phone",
  ],
  "message": Array [
    "name: Michele",
  ],
  "name": "Michele",
}
`;

exports[`validate3 1`] = `
Object {
  "email": "michele@sciabarra.com",
  "errors": Array [
    "michele@sciabarra.com is not an email address",
  ],
  "message": Array [
    "name: Michele",
    "phone: 1234567890",
  ],
  "name": "Michele",
  "phone": "1234567890",
}
`;
